src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -529,7 +667,61 @@ private Map<String, PendingImportObject<BrAPITrial>> initialize_trialByNameNoSco
             throw new InternalServerException(e.toString(), e);
         }
     }
+    private String simpleStudyName(String scopedName){
+        return scopedName.replaceFirst(" \\[.*\\]", "");
+    }
 
+    /**
+     * Converts year String to SeasonDbId
+     * <br>
+     * NOTE: This assumes that the only Season records of interest are ones
+     * with a blank name or a name that is the same as the year.
+     *
+     * @param years this only looks at the first year of the list.
+     * @param programId the program ID.
+     * @return the DbId of the season-record associated with the first year
+     * of the 'years' list (see NOTE above)
+     */
+    private String yearsToSeasonDbId(List<String> years, UUID programId) {
+        String year = years.get(0);
+        String dbID = null;
+        if (this.yearToSeasonDbIdCache.containsKey(year) ){ // get it from cache if possable
+            dbID = this.yearToSeasonDbIdCache.get(year);
+        }
+        else{
+            dbID = this.yearToSeasonDbIdFromDatabase(year,programId);
+            this.yearToSeasonDbIdCache.put(year, dbID);
+        }
+        return dbID;
+    }
 
+    private String yearToSeasonDbIdFromDatabase(String year, UUID programId) {
+        BrAPISeason targetSeason = null;
+        List<BrAPISeason> seasons;
+        try {
+            seasons = this.brAPISeasonDAO.getSeasonByYear(year, programId);
+            for( BrAPISeason season : seasons){
+                if(null == season.getSeasonName() || season.getSeasonName().isBlank() || season.getSeasonName().equals(year)){
+                    targetSeason = season;
+                    break;
+                }
+            }
+            if (targetSeason == null){
+                BrAPISeason newSeason = new BrAPISeason();
+                newSeason.setYear( Integer.parseInt(year) );
+                newSeason.setSeasonName( year );
+                targetSeason = this.brAPISeasonDAO.addOneSeason( newSeason, programId );
+            }
+
+        } catch (ApiException e) {
+            log.error(e.getResponseBody(), e);;
+        }
 
+        String seasonDbId = (targetSeason==null) ? null : targetSeason.getSeasonDbId();
+        return seasonDbId;
+    }
+
+    private boolean isRefSource(BrAPIExternalReference brAPIExternalReference) {
```

Comment:
I'd recommend either renaming this method to `isTrialRefSource` or add an additional parameter to specify what `ExternalReferenceSource` to append.